velox/common/time/CpuWallTimer.h

```diff
@@ -57,4 +57,32 @@ class CpuWallTimer {
   CpuWallTiming& timing_;
 };
 
+// Adds elapsed CPU and wall time to a CpuWallTiming.
+// Uses copy of the timing and a functional to perform an action of the updated
+// timing upon destruction.
+template <typename F>
+class CpuWallTimerCopy {
+ public:
+  explicit CpuWallTimerCopy(const CpuWallTiming& timing, F&& func)
```

Comment:
Can we just extend CpuWallTimer? F doesn't need to be template?
```
class CpuWallTimer {
   using Callback = std::function<void(const CpuWallTiming&)>;
   ...
  ~CpuWallTimer() {
       timing_.cpuNanos += process::threadCpuNanos() - cpuTimeStart_;
       const auto duration = std::chrono::duration_cast<std::chrono::nanoseconds>(std::chrono::steady_clock::now() - wallTimeStart_);
      timing_.wallNanos += duration.count();
      if (callback_ != nullptr) {
         callback_(timing_)
      }
  }
```

Reply:
@xiaoxmeng
The current CpuWallTimer cannot be used because it requires reference to the 'Timing', which we cannot give as we protect it with the folly::synchronized.
Instead of polluting the existing class, I decided to create a new small one.

I tried to use the typed std::function() argument and class member (not template), but somehow couldn't make it compile with the lambda supplied at the Timer construction time.
The templated variant is the one that worked for me.

Reply:
Can we consider to define the class as
```
DeltaCpuWallTimer(F& timerUpdateFn) which doesn't need to access the source wall time?
```